galera/src/replicator_str.cpp

```diff
@@ -994,6 +1011,17 @@ ReplicatorSMM::request_state_transfer (void* recv_ctx,
     // in the sst_received.
     sst_state_ = SST_WAIT;
 
+    // We need to release the mutex before sending the request. This is
+    // necessary to avoid the server hang in case of an SST failure.
+    //
+    // Server hang is possible in below case
+    //
+    // - When donor is killed during SST, sst_joiner_thread shall be
```

Comment:
Is my understanding correct that here we are talking about killing donor before initial sst phase (waiting for sst-meta-info transfer)? When we receive sst-data and donor is killed I think we just get end of stream and then handle partially sent SST in another way, right?
If yes, could you make comments more accurate?

Reply:
Yes. You're right!! It is handled in a different way. For this, I'm thinking to change the comments to 
```
    // We need to release the mutex before sending the request. This is
    // necessary to avoid the server hang in case of a failure during the
    // initial stage of SST.
    //
    // Server hang is possible in below case
    //
    // - When donor is killed during the initial stage of SST (during metadata
    //   transfer), the sst_joiner_thread shall be waiting for the sst_mutex_
    //   in sst_recieved().
    // - Applier thread shall be holding the sst_mutex_ and be waiting
    //   for the state transfer to finish.
```
Please let me know if this fine or it sill needs more explanation.